tests/antares/services/api_services/test_study_api.py

```diff
@@ -330,3 +335,86 @@ def test_create_link_same_area(self):
             match=f"Could not create the link {area} / {area}: A link cannot start and end at the same area",
         ):
             self.study.create_link(area_from=area, area_to=area)
+
+    def test_run_and_wait_antares_simulation(self):
+        parameters = AntaresSimulationParameters(solver=Solver.COIN, nb_cpu=2, unzip_output=True, presolve=False)
+
+        # patch simulates the repeating intervals so that we don't have to wait X seconds during the tests
+        with requests_mock.Mocker() as mocker, patch("time.sleep", return_value=None):
+            run_url = f"https://antares.com/api/v1/launcher/run/{self.study_id}"
+            job_id = "1234-g6z17"
+            mocker.post(run_url, json={"id": job_id}, status_code=200)
+
+            job_url = f"https://antares.com/api/v1/launcher/jobs/{job_id}"
+            output_id = "2024abcdefg-output"
+
+            # mock can take a list of responses that it maps one by one to each request, since we're doing multiple ones
+            response_list = [
+                {
+                    "json": {
+                        "id": job_id,
+                        "status": "pending",
+                        "launcher_params": dumps(parameters.to_api()),
+                    },
+                    "status_code": 200,
+                },
+                {
+                    "json": {
+                        "id": job_id,
+                        "status": "running",
+                        "launcher_params": dumps(parameters.to_api()),
```

Comment:
You can store `dumps(parameters.to_api())` inside a variable instead of calling it 4 times